packages/checkout/sdk/src/api/http/httpClient.ts

```diff
@@ -0,0 +1,110 @@
+import axios, { AxiosInstance, AxiosRequestConfig, AxiosResponse } from 'axios';
+import { CheckoutModuleConfiguration } from '../../types';
+import { CheckoutError, CheckoutErrorType } from '../../errors';
+
+const PUBLISHABLE_KEY_PREFIX = 'pk_imapik-';
+
+const publishableKeyWhitelistedDomain = [
+  'https://checkout-api.dev.immutable.com',
+  'https://checkout-api.sandbox.immutable.com',
+  'https://checkout-api.immutable.com',
+];
+
+export class HttpClient {
+  protected config: any;
+
+  protected axiosInstance: AxiosInstance;
+
+  constructor(config?: CheckoutModuleConfiguration) {
+    this.config = config;
+
+    this.axiosInstance = axios.create();
+    this.setupInterceptors();
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  private shouldAddPublishableKey(url: string): boolean {
+    // Check if the request is going to a whitelisted domain
+    return publishableKeyWhitelistedDomain.some((domain) => url.startsWith(domain));
+  }
+
+  private setupInterceptors() {
+    this.axiosInstance.interceptors.request.use(
+      (config) => {
+        const publishableKey = this.config.baseConfig?.publishableKey;
+        if (publishableKey && this.shouldAddPublishableKey(config.url || '')) {
+          if (!publishableKey.startsWith(PUBLISHABLE_KEY_PREFIX)) {
+            throw new Error(
+              'Invalid Publishable key. Create your Publishable key in Immutable developer hub.'
+              + ' https://hub.immutable.com',
+            );
+          }
+
+          // Add publishable key header for whitelisted domains
+          // eslint-disable-next-line no-param-reassign
+          config.headers['x-immutable-publishable-key'] = publishableKey;
```

Comment:
Is the reason for not using
```
axios.defaults.headers.common['x-immutable-publishable-key'] = publishableKey;
```
and therefore avoid to wrap the `get`, `post`, ... requests the fact that you are injecting the header only for some domains?

Reply:
Yep exactly, otherwise those global headers would apply to OrderBook and anything else in the SDK that uses axios.